<?xml version="1.0"?>
<!-- 
    Data fuser launch file
 -->
<launch>
    <arg name="load_params"             default="false"/>
    
    <arg name="rate"                    default="15.0"/>
    <arg name="anon"                    default="True"/>
    <arg name="reset"                   default="True"/>

    <arg name="namespace"               default="/vpr_nodes"/>

    <arg name="log_level"               default="2"/>
    <arg name="param_file"              default="sim_cfg.yaml"/>
    <arg name="ID"                      default="0"/>

    <group if="$(arg load_params)">
        <!-- Load in params; must be done outside of a namespace -->
        <rosparam command="load" file="$(find aarapsi_robot_pack)/cfg/$(arg param_file)"/>
    </group>
    
    <group ns="$(arg namespace)">   
        <!-- Data Fuser Node: -->
        <node pkg="aarapsi_robot_pack" type="data_fuser.py" name="data_fuser"  output="screen"
                args="
                        -N data_fuser -r $(arg rate) -a $(arg anon)
                        -V $(arg log_level) -R $(arg reset) -ID $(arg ID)" />
    </group>

</launch>
